variables.tf

```diff
@@ -140,24 +140,7 @@ variable "maintenance_window" {
     start_hour   = optional(number, 0)
     start_minute = optional(number, 0)
   })
-  default = {
-    day_of_week  = 0
-    start_hour   = 0
-    start_minute = 0
-  }
-
-  validation {
-    condition     = var.maintenance_window.day_of_week >= 0 && var.maintenance_window.day_of_week <= 6
-    error_message = "Invalid maintenance_window.day_of_week value"
-  }
-  validation {
-    condition     = var.maintenance_window.start_hour >= 0 && var.maintenance_window.start_hour <= 23
-    error_message = "maintenance_window.start_hour must be between 0 and 23"
-  }
-  validation {
-    condition     = var.maintenance_window.start_minute >= 0 && var.maintenance_window.start_minute <= 59
-    error_message = "maintenance_window.start_minute must be between 0 and 59"
-  }
```

Comment:
This all seems like useful validation logic, it would be a shame to lose it. Is the problem that we're using a null default and we just need to wrap this in a null check? There are a couple ways you might approach this....

[Coalesce](https://developer.hashicorp.com/terraform/language/functions/coalesce):

```hcl
  validation {
    condition     = coalesce(var.maintenance_window.day_of_week, 0) >= 0 && coalesce(var.maintenance_window.day_of_week, 0) <= 6 
    error_message = "Invalid maintenance_window.day_of_week value"
  }
```

Ternary expression:

```hcl
  validation {
    condition     = var.maintenance_window == null ? true : var.maintenance_window.day_of_week >= 0 && var.maintenance_window.day_of_week <= 6 
    error_message = "Invalid maintenance_window.day_of_week value"
  }
```

Reply:
The "coalesce" option has the same problem actually, but the ternary operator seems to work.

Reply:
I agree in this case the plan catches the error, but there are many (most?) cases where the plan doesn't catch an out-of-range input. IMO we should settle on a method as a practice and teach folks how to do these validations. 
Complicated validations always trip me and I've settled on the ternary method, as that makes the most sense to me and seems to work in all the cases where I've had the most trouble with these validations.

Reply:
I'm good with removal since the provider handles validation.